athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/data/BlockUtils.java

```diff
@@ -464,6 +491,24 @@ public static int copyRows(Block srcBlock, Block dstBlock, int firstRow, int las
         return rowsCopied;
     }
 
+    /**
+     * We will convert any types that are not supported by setValue to types that are supported
+     * ex) (not supported) org.joda.time.LocalDateTime which is returned on read from vectors
+     * will be converted to (supported) java.time.ZonedDateTime
+     * @param o1 object with unsupported type
+     * @return another object with supported/converted type
+     */
```

Comment:
why can't we just make setvalue handle this so we avoid the extra type checking and branching?